src/vstest.console/InProcessVsTestConsoleWrapper.cs

```diff
@@ -839,7 +840,47 @@ public Task InitializeExtensionsAsync(IEnumerable<string> pathToAdditionalExtens
         ITestRunAttachmentsProcessingEventsHandler eventsHandler,
         CancellationToken cancellationToken)
     {
-        throw new NotImplementedException();
+        _testPlatformEventSource.TranslationLayerTestRunAttachmentsProcessingStart();
+
+        try
+        {
+            var attachmentProcessingPayload = new TestRunAttachmentsProcessingPayload
+            {
+                Attachments = attachments,
+                InvokedDataCollectors = invokedDataCollectors,
+                RunSettings = processingSettings,
+                CollectMetrics = collectMetrics
+            };
+
+            using (cancellationToken.Register(() =>
+                TestRequestManager?.CancelTestRunAttachmentsProcessing()))
+            {
+                // Awaiting the attachment processing task here. The implementation of the
+                // underlying operation guarantees the event handler is called when processing
+                // is complete, so when awaiting ends, the results have already been passed to
+                // the caller via the event handler. No need for further synchronization.
+                await Task.Run(() =>
+                        TestRequestManager?.ProcessTestRunAttachments(
+                            attachmentProcessingPayload,
+                            new InProcessTestRunAttachmentsProcessingEventsHandler(eventsHandler),
+                            new ProtocolConfig { Version = _highestSupportedVersion }),
+                        CancellationToken.None)
```

Comment:
This is in sync with the original out-of-process implementation of the attachment processing logic where a separate call would set the cancellation token. The implementation of the `ProcessTestRunAttachments` checks the cancellation token before certain steps, and if the cancellation token suddenly got set, the operation as a whole is successfully cancelled. I agree that we can pass the cancellation token to the Task.Run call though, as an optimization, so that we don't start the task at all if the token is set before getting here.

Reply:
Backtracking on this. We need to go back to passing `CancellationToken.None` to the `Task.Run` call. We need to make sure the cancel event is correctly propagated to the caller (otherwise the caller could be left hanging), and this only happens in the internal attachment processing logic, so we need to make sure we get there. Cancelling the task before it starts here would mean we need to duplicate that logic here and this will make it hard to maintain.

Reply:
Could you please add this comment in the code. This is definitely something that I would consider refactoring if I don't see any explanation. Also, do we have a test (or could we add one) that would ensure that if passing the cancellation token we have some check failing? This would provide belt and braces solution for us not to regress.

Reply:
Added a comment as you suggested, thanks ! Unfortunately though, we cannot add a test to make sure we don't pass the wrong cancellation token since Task.Run is not mockable and we cannot verify the arguments it receives.